File: Android Architecture/PraciticeRecycleView/app/src/main/java/com/practice/praciticerecycleview/MyAdapter.java
package com.practice.praciticerecycleview;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.MyViewHolder> {

    String name[];
    String description[];
    int images[];
    Context context;

    public MyAdapter(Context context,  String s1[], String s2[], int images[]){
        this.context=context;

        description =s2;
        name =s1;
        this.images=images;

    }


    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        LayoutInflater inflater = LayoutInflater.from(context);
        View view =  inflater.inflate(R.layout.my_row, parent,false);
        //Tak zwracamy ten layout my row  do MyViewHolder (klasa nizej)
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {


    holder.myText1.setText(name[position]);
    holder.myText2.setText(description[position]);
    holder.imageView.setImageResource(images[position]);

    }

    @Override
    public int getItemCount() {
        return name.length;
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {

        TextView myText1, myText2;
        ImageView imageView;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);

            myText1 = itemView.findViewById(R.id.myTitle);
            myText2 = itemView.findViewById(R.id.myDesc);
            imageView = itemView.findViewById(R.id.myImg);

        }
    }
}

File: Android Architecture/Room/app/src/main/java/com/practice/room/fragments/list/MyAdapter.kt
package com.practice.room.fragments.list

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.findNavController
import androidx.recyclerview.widget.RecyclerView
import com.practice.room.R
import com.practice.room.model.User
import kotlinx.android.synthetic.main.my_row.view.*

class MyAdapter : RecyclerView.Adapter<MyAdapter.MyViewHolder>() {

    private var userList = emptyList<User>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        return MyViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.my_row, parent,false))
    }

    override fun getItemCount(): Int {
        return userList.size
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {

        val currentItem = userList[position]
        holder.itemView.myId.text = currentItem.id.toString();
        holder.itemView.myFirstname.text= currentItem.firstName
        holder.itemView.myLastname.text= currentItem.secondName
        holder.itemView.myAge.text= currentItem.age.toString()

        holder.itemView.rowLayout.setOnClickListener {
            val actrion = ListFragmentDirections.actionListFragmentToUpdateFragment(currentItem)
            holder.itemView.findNavController().navigate(actrion)
        }
    }


    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    }


    fun setData(user: List<User>){
        this.userList=user
        //uzywa,y to nizej bo array moze sie zmienic etc 
        notifyDataSetChanged()
    }


}